spec/controllers/api/v1/accounts/custom_filters_controller_spec.rb

```diff
@@ -65,23 +95,18 @@
     end
 
     context 'when it is an authenticated user' do
-      let(:user) { create(:user, account: account) }
-      let(:new_account) { create(:account) }
-      let(:new_user) { create(:user, account: new_account) }
-
       it 'creates the filter' do
-        expect do
-          post "/api/v1/accounts/#{account.id}/custom_filters", headers: user.create_new_auth_token,
-                                                                params: payload
-        end.to change(CustomFilter, :count).by(1)
+        post "/api/v1/accounts/#{account.id}/custom_filters", headers: user.create_new_auth_token,
+                                                              params: payload
 
         expect(response).to have_http_status(:success)
         json_response = response.parsed_body
         expect(json_response['name']).to eq 'vip-customers'
+        expect(json_response['count']).to eq '1'
       end
 
       it 'gives the error for 51st record' do
-        CustomFilter::MAX_FILTER_PER_USER.times do
+        (CustomFilter::MAX_FILTER_PER_USER - 1).times do
```

Comment:
Rather than doing the ` -1`. let's destroy all the filters above this or do an account.custom_filters count as a query, which will be more readable  and won't let the reader wonder about the logic behind `-1`